### PR TITLE
chore(Update): update to alpha.47

### PR DESCRIPTION
**Pull request by NathanWalker**
Simplifying things due to improvements with latest alpha.47.

Also added ability to stop/restart the stream.

### CODE REVIEW

File: src/app/app.ts [hunk 1]
```diff
@@ -26,17 +24,33 @@ class AppComponent {
...
-    this.pusher.unsubscribe(btoa(channel));
+  public toggleSearch(channel) {
+    for (let ch of this.channels) {
+      if (ch.term === channel.term) {
+        if (ch.active) {
+          ch.active = false;
+          this.pusher.unsubscribe(btoa(channel.term));
+        } else {
+          ch.active = true;
+          var encoded = btoa(channel.term);
```
Comment (jackfranklin) [inline src/app/app.ts]: `var` => `const` ?

File: src/app/subscription.ts [hunk 2]
```diff
@@ -1,46 +1,42 @@
...
+    });
   }
 
-  private newTweet(data: Object) {
+  newTweet(data: Object) {
     this.tweets.push(data);
   }
 
-  public onDestroy() {
+  ngOnDestroy() {
```
Comment (jackfranklin) [inline src/app/subscription.ts]: deleting the interfaces and then renaming the methods - is this a change in Angular?

**Comment by jackfranklin** [comment]
Thanks! Couple of comments but otherwise looks great :)

File: src/app/app.ts (hunk 1 above)
Comment (NathanWalker) [inline src/app/app.ts]: If anything maybe should be: `var` => `let`.

I'll push something even better in moment though.

File: src/app/subscription.ts (hunk 2 above)
Comment (NathanWalker) [inline src/app/subscription.ts]: The interfaces were fine either way I believe, but the methods got renamed in .47:

https://github.com/angular/angular/commit/604c8bbad54ffbcec1d6a5e407821adda8a74a19

File: src/app/app.ts [hunk 3]
```diff
@@ -26,17 +24,32 @@ class AppComponent {
...
-    this.pusher.unsubscribe(btoa(channel));
+  public toggleSearch(channel) {
+    for (let ch of this.channels) {
+      if (ch.term === channel.term) {
+        if (ch.active) {
+          ch.active = false;
+          this.pusher.unsubscribe(btoa(channel.term));
+        } else {
+          ch.active = true;
+          this.pusher.subscribe(btoa(channel.term));
```
Comment (NathanWalker) [inline src/app/app.ts]: Better to have no ref at all :+1:

Comment (jackfranklin) [inline src/app/app.ts]: :+1:

File: src/app/subscription.ts (hunk 2 above)
Comment (jackfranklin) [inline src/app/subscription.ts]: in which case, can we keep the interfaces?

Comment (NathanWalker) [inline src/app/subscription.ts]: We can keep them, I'll push a commit in a minute with them back. Fixing the start/restart stream as that wasn't working quite right. I may push without the toggling of the stream until I have time to refactor a bit as we could bring a service in the mix to make things a bit more seamless there. Seeing what I can do.

**Comment by NathanWalker** [comment]
@jackfranklin Ok updated with the toggling of the stream working now :+1:  and added back the interfaces.

For some reason `ngOnChanges` is not firing like it should. Not sure if this is something with .47 or not. You may be able to investigate further. In meantime, I left comments to move a block inside `ngAfterViewChecked` into `ngOnChanges` (which is commented out with comments) when we can get it working again.

**Comment by NathanWalker** [comment]
Basically `ngOnChanges` should fire everytime the `channel` is modified in `AppComponent` but it appears to only fire once when the `SubscriptionComponent` is created and the bindings are set but never again. For now, I was able to suffice by picking up the binding changes in `ngAfterViewChecked` for now, however it would be ideal to have that done in `ngOnChanges`. 

Another way might be to have a `PusherService` which would use an `EventEmitter` to notify the components of such changes and have `AppComponent` and `SubscriptionComponent` subscribe to the `EventEmitter` in that `PusherService` to get the changes. But again, `ngOnChanges` should fire anytime the bindings change I believe unless there's something I'm missing.

**Comment by NathanWalker** [comment]
Everything is working nicely though with this changeset so feel free to merge and we can continue to improve with subsequent PR's down the road :+1:

File: src/app/app.ts [hunk 4]
```diff
@@ -26,17 +24,29 @@ class AppComponent {
...
-  public stopSearch(channel) {
-    this.pusher.unsubscribe(btoa(channel));
+  public toggleSearch(channel) {
+    for (let ch of this.channels) {
+      if (ch.term === channel.term) {
+        if (ch.active) {
+          ch.active = false;
+        } else {
+          ch.active = true;
+        }
```
Comment (jackfranklin) [inline src/app/app.ts]: you could tidy this up with `ch.active = !ch.active;`

**Comment by jackfranklin** [comment]
@NathanWalker thank you! Those ideas sound good, let's keep incrementing slowly with PRs. Appreciate your help. I've got one teeny comment and then let's merge this in.

Also if you want to send your address & t-shirt size to jack at pusher dot com we'll send you some swag as a small thank you :)

**Comment by NathanWalker** [comment]
@jackfranklin Got your comment addressed, good feedback! :+1:

**Comment by jackfranklin** [comment]
@NathanWalker thanks!

**Pull request merged**